packages/adapter-neon/src/conversion.ts

```diff
@@ -233,29 +75,35 @@ export function fieldToColumnType(fieldTypeId: number): ColumnType {
     case ArrayColumnType.INET_ARRAY:
     case ArrayColumnType.CIDR_ARRAY:
     case ArrayColumnType.XML_ARRAY:
-      return ColumnTypeEnum.TextArray
+      return ok(ColumnTypeEnum.TextArray)
     case ArrayColumnType.DATE_ARRAY:
-      return ColumnTypeEnum.DateArray
+      return ok(ColumnTypeEnum.DateArray)
     case ArrayColumnType.TIME_ARRAY:
-      return ColumnTypeEnum.TimeArray
+      return ok(ColumnTypeEnum.TimeArray)
     case ArrayColumnType.TIMESTAMP_ARRAY:
-      return ColumnTypeEnum.DateTimeArray
+      return ok(ColumnTypeEnum.DateTimeArray)
     case ArrayColumnType.JSON_ARRAY:
     case ArrayColumnType.JSONB_ARRAY:
-      return ColumnTypeEnum.JsonArray
+      return ok(ColumnTypeEnum.JsonArray)
     case ArrayColumnType.BYTEA_ARRAY:
-      return ColumnTypeEnum.BytesArray
+      return ok(ColumnTypeEnum.BytesArray)
     case ArrayColumnType.UUID_ARRAY:
-      return ColumnTypeEnum.UuidArray
+      return ok(ColumnTypeEnum.UuidArray)
     case ArrayColumnType.INT8_ARRAY:
     case ArrayColumnType.OID_ARRAY:
-      return ColumnTypeEnum.Int64Array
+      return ok(ColumnTypeEnum.Int64Array)
     default:
-      if (fieldTypeId >= 10000) {
-        // Postgres Custom Types
-        return ColumnTypeEnum.Enum
+      // Postgres Custom Types
+      if (fieldType.id > maxSystemCatalogOID) {
+        if (fieldType.name !== undefined && ['citext', 'ltree', 'lquery', 'ltxtquery'].includes(fieldType.name)) {
```

Comment:
The list simply mirrors https://github.com/sfackler/rust-postgres/blob/10edbcb46c44933417e8d2e7a1c1d63c4119beb3/postgres-types/src/lib.rs#L707-L710, not actually tested for any types except `citext`.